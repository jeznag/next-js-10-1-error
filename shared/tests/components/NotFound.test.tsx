import React from 'react'
import renderer from 'react-test-renderer'
import { NotFound } from '@components'
import '../__mocks__/match-media'

describe('NotFound', () => {
    it('should render correctly with a 404 error', () => {
        const tree = renderer
            .create(<NotFound code='404' />)
            .toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('should render correctly with a 401 error', () => {
        const tree = renderer
            .create(<NotFound code='401' />)
            .toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('should render correctly with a 418 error', () => {
        const tree = renderer
            .create(<NotFound code='418' />)
            .toJSON()

        expect(tree).toMatchSnapshot()
    })
})
