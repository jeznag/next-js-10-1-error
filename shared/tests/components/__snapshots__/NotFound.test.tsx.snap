// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render correctly with a 401 error 1`] = `
<div
  className="ant-row ant-row-center ant-row-middle bg-white text-center"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <div
    className="sc-gsTCUz bWQjOv"
  >
    <h1
      className="sc-dlfnbm JZVZa text-error mb-0"
    >
      401
    </h1>
    <h6
      className="mb-1 mt-1 text-body"
    >
      Unauthorized
       
    </h6>
    <br />
    <a
      className="ant-btn"
      href="/logout"
      onClick={[Function]}
    >
      <span>
        Back to login
      </span>
    </a>
  </div>
</div>
`;

exports[`NotFound should render correctly with a 404 error 1`] = `
<div
  className="ant-row ant-row-center ant-row-middle bg-white text-center"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <div
    className="sc-gsTCUz bWQjOv"
  >
    <h1
      className="sc-dlfnbm JZVZa text-error mb-0"
    >
      404
    </h1>
    <h6
      className="mb-1 mt-1 text-body"
    >
      This page could not be found
       
    </h6>
    <br />
  </div>
  <div
    className="sc-bdfBwQ eZppOP"
  >
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
  </div>
</div>
`;

exports[`NotFound should render correctly with a 418 error 1`] = `
<div
  className="ant-row ant-row-center ant-row-middle bg-white text-center"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <div
    className="sc-gsTCUz bWQjOv"
  >
    <h1
      className="sc-dlfnbm JZVZa text-error mb-0"
    >
      418
    </h1>
    <h6
      className="mb-1 mt-1 text-body"
    >
      I'm a teapot
       
    </h6>
    <br />
  </div>
  <div
    className="sc-bdfBwQ eZppOP"
  >
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
  </div>
</div>
`;
